pkg/nsx/services/ipblocksinfo/ipblocksinfo.go

```diff
@@ -172,6 +226,40 @@ func (s *IPBlocksInfoService) createOrUpdateIPBlocksInfo(ctx context.Context, ip
 	return nil
 }
 
+func (s *IPBlocksInfoService) getSharedSubnetsCIDRs(vpcConfigList []v1alpha1.VPCNetworkConfiguration) (externalIPCIDRs []string, privateTGWIPCIDRs []string, err error) {
+	sharedSubnet := sets.New[string]()
+	for _, vpcConfigCR := range vpcConfigList {
+		for _, subnet := range vpcConfigCR.Spec.Subnets {
+			sharedSubnet.Insert(subnet)
+		}
+	}
+	for _, subnetPath := range sharedSubnet.UnsortedList() {
+		vpcInfo, err := common.ParseVPCResourcePath(subnetPath)
+		if err != nil {
+			log.V(1).Error(fmt.Errorf("failed to parse VPC resource path: %w", err), "path", subnetPath)
+			continue
+		}
+		associate := fmt.Sprintf("%s:%s:%s", vpcInfo.ProjectID, vpcInfo.VPCID, vpcInfo.ID)
+		subnet, err := s.subnetService.GetNSXSubnetFromCacheOrAPI(associate)
```

Comment:
Is it possible the ipAddress in NSX Subnet got updated, do we need subnet service to notify ipblockinfo?

Reply:
It's shared subnet. It seems that current share subnet didn't implement the update operation.